comps/guardrails/src/hallucination_detection/Dockerfile

```diff
@@ -20,7 +20,8 @@ RUN pip install --no-cache-dir --upgrade pip setuptools && \
       pip install --no-cache-dir --extra-index-url https://download.pytorch.org/whl/cpu -r /home/user/comps/guardrails/src/hallucination_detection/requirements.txt; \
     else \
       pip install --no-cache-dir -r /home/user/comps/guardrails/src/hallucination_detection/requirements.txt; \
-    fi
+    fi && \
+    (python -m pip list | grep -i 'nvidia' | awk '{print $1}' | xargs -r -L1 python -m pip uninstall -y || true)
```

Comment:
No need to remove otherwise `non-cpu` container builds will not be functional due to removing essential dependencies.
We just want to make sure the CPU builds are clean.